Add UseeLink 302EU-Zigbee power strip (_TZ3000_cfnprab5)

hangy opened the pull request:
Adds support for UseeLink 302EU-Zigbee, based on https://github.com/zigpy/zha-device-handlers/issues/1632#issuecomment-1627072807

kender65 commented:
Hi!
Based on the description "UseeLink 302EU-Zigbee" I guess [this](https://www.amazon.de/Steckdosenleiste-UseeLink-Mehrfachsteckdose-Unabh%C3%A4ngig-Anschl%C3%BCsse-302EU-Zigbee/dp/B0CTMXYQLZ?th=1) is the device in question.
Wouldn't it be better to name the class "Plug_4AC_**4**USB_TZ3000_cfnprab5" to make it clear, that it has 4 USB (3 x USB-A, 1 x USB-C) ports?
I got this [SM-302Ez-PA20](https://www.amazon.de/UseeLink-Steckdosenleiste-AC-Anschl%C3%BCsse-Schnellladeger%C3%A4t-Sprachsteuerung/dp/B0CT8FXWF5?source=ps-sl-shoppingads-lpcontext&ref_=fplfs&psc=1&smid=A2M76ZDYXTOLGU) which looks nearly the same. But I think mine has the ability to set "child_lock" and at least to control the "power_outage_memory" (or "power_on_state") as mentioned [here](https://github.com/zigpy/zha-device-handlers/issues/1632#issuecomment-1213118577). Initially I integrated this power strip with Z2M on HAOS although not listed as a supported device. Pairing went well but surprisingly it showed up as [SM-0306E-2W](https://www.zigbee2mqtt.io/devices/SM-0306E-2W.html) which caused the wrong picture of the device on the Z2M UI. I was able to toggle the outlets and USB ports like in ZHA and additionally could set the "power_outage_memory" to it's expected values "on", "off", "restore" from the Z2M UI and it worked as expected although I got an error message for an "UNSUPPORTED_ATTRIBUTE". The same goes for the "child_lock". Setting from the UI to both values worked as expected despite an "UNSUPPORTED_ATTRIBUTE" message. I think the behaviour matches the [comment](https://github.com/zigpy/zha-device-handlers/issues/1632#issuecomment-1213380426) from @javicalle as the "power_outage_memory" was only visible  for endpoint 1 (and as general attribute which may be forced the error message).

To figure out what was the problem I tried the other way with ZHA, which I haven't used until then.
I got it working with your quirk, thanks for that. Toggling the 4 AC outlets and the USB ports as a group from the ZHA UI and / or from the developer tools works without issues.
![image](https://github.com/user-attachments/assets/e511ba1d-b236-4b31-adb1-4f29098b77ba)
Further configuration for "power_on_state", "power_outage_memory" or "start_up_on_off", or whatever you would call it, doesn't work so I extended your quirk with "quirk_id = TUYA_PLUG_ONOFF" in the hope of getting "child_lock" and "power_outage_memory" / "power_on_state" working. I got additional attributes in the "TuyaZBOnOffAttributeCluster"
![image](https://github.com/user-attachments/assets/16585118-5dbb-46ba-9860-d4b0eb52fa75)
but haven't been able to get it working. I couldn't set the value to "on", "off" or "restore". Initially it has the value "None".
The same is for
![image](https://github.com/user-attachments/assets/3c36acd0-181c-4584-86d5-bfef01535648)
Any idea how to get it working, for at least endpoint 1?

hangy commented:
> Hi!

🙋‍♂️

> Based on the description "UseeLink 302EU-Zigbee" I guess [this](https://www.amazon.de/Steckdosenleiste-UseeLink-Mehrfachsteckdose-Unabh%C3%A4ngig-Anschl%C3%BCsse-302EU-Zigbee/dp/B0CTMXYQLZ?th=1) is the device in question. 

That's the one.

> Wouldn't it be better to name the class "Plug_4AC_**4**USB_TZ3000_cfnprab5" to make it clear, that it has 4 USB (3 x USB-A, 1 x USB-C) ports?

Fair point! Maybe just `_USB_` or `_1USB_`, as all USB ports can _only_ be switched on or off as one group? I haven't looked at that many quirks, yet, so I'm not aware of a naming convention for a case like this.

> I got this [SM-302Ez-PA20](https://www.amazon.de/UseeLink-Steckdosenleiste-AC-Anschl%C3%BCsse-Schnellladeger%C3%A4t-Sprachsteuerung/dp/B0CT8FXWF5?source=ps-sl-shoppingads-lpcontext&ref_=fplfs&psc=1&smid=A2M76ZDYXTOLGU) which looks nearly the same. But I think mine has the ability to set "child_lock" and at least to control the "power_outage_memory" (or "power_on_state") as mentioned [here](https://github.com/zigpy/zha-device-handlers/issues/1632#issuecomment-1213118577). Initially I integrated this power strip with Z2M on HAOS although not listed as a supported device. Pairing went well but surprisingly it showed up as [SM-0306E-2W](https://www.zigbee2mqtt.io/devices/SM-0306E-2W.html) which caused the wrong picture of the device on the Z2M UI. I was able to toggle the outlets and USB ports like in ZHA and additionally could set the "power_outage_memory" to it's expected values "on", "off", "restore" from the Z2M UI and it worked as expected although I got an error message for an "UNSUPPORTED_ATTRIBUTE". The same goes for the "child_lock". Setting from the UI to both values worked as expected despite an "UNSUPPORTED_ATTRIBUTE" message. I think the behaviour matches the [comment](https://github.com/zigpy/zha-device-handlers/issues/1632#issuecomment-1213380426) from @javicalle as the "power_outage_memory" was only visible for endpoint 1 (and as general attribute which may be forced the error message).
> 
> To figure out what was the problem I tried the other way with ZHA, which I haven't used until then. I got it working with your quirk, thanks for that. Toggling the 4 AC outlets and the USB ports as a group from the ZHA UI and / or from the developer tools works without issues. ![image](https://private-user-images.githubusercontent.com/94765252/383118428-e511ba1d-b236-4b31-adb1-4f29098b77ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1Mzg0NzEsIm5iZiI6MTczMTUzODE3MSwicGF0aCI6Ii85NDc2NTI1Mi8zODMxMTg0MjgtZTUxMWJhMWQtYjIzNi00YjMxLWFkYjEtNGYyOTA5OGI3N2JhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDIyNDkzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzMDk3ZGQ1ZTI1YmRkNmMwNWJmZDAyMTY4ZWJkZDc2ODZjNDJlNGNhOWM3ZDg2ZGFmZmNlZDJhOWE3ZGZmMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8V8mox8Q8hu1Lcb_64YuoFDd-quSoLohpB-ZNmB-AWQ) Further configuration for "power_on_state", "power_outage_memory" or "start_up_on_off", or whatever you would call it, doesn't work so I extended your quirk with "quirk_id = TUYA_PLUG_ONOFF" in the hope of getting "child_lock" and "power_outage_memory" / "power_on_state" working. I got additional attributes in the "TuyaZBOnOffAttributeCluster" ![image](https://private-user-images.githubusercontent.com/94765252/383117731-16585118-5dbb-46ba-9860-d4b0eb52fa75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1Mzg0NzEsIm5iZiI6MTczMTUzODE3MSwicGF0aCI6Ii85NDc2NTI1Mi8zODMxMTc3MzEtMTY1ODUxMTgtNWRiYi00NmJhLTk4NjAtZDRiMGViNTJmYTc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDIyNDkzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmZDIxM2YzZjUyMjk2ZjQ1YzBkNTMwZDE5NjhhMmU2NGYxOWUxYWQzMGRkOWQ4MThkZjEwMGNhODlmN2JlNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zvd0DVDsJ-hKa-DYbP8zUR21PXIINasSNh6h2GEQYNA) but haven't been able to get it working. I couldn't set the value to "on", "off" or "restore". Initially it has the value "None". The same is for ![image](https://private-user-images.githubusercontent.com/94765252/383113359-3c36acd0-181c-4584-86d5-bfef01535648.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1Mzg0NzEsIm5iZiI6MTczMTUzODE3MSwicGF0aCI6Ii85NDc2NTI1Mi8zODMxMTMzNTktM2MzNmFjZDAtMTgxYy00NTg0LTg2ZDUtYmZlZjAxNTM1NjQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDIyNDkzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2NmY0NDU2ZDMzN2Q0YzNiNjc4ZGM5OTJjZGQ1M2UyYTA5YzUwNDU0ZTE1M2VkNjdiNDEyNDk5MDY5NjY4ZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.51vOFaG2xyJdWGZMNjoa5e-DtyW1t4a70cRw3Lym5Uw) Any idea how to get it working, for at least endpoint 1?

I'm not too experience with zigpy or ZHA, so I guess we're both figuring it out in parallel. 😉 
For my device, the "Startverhalten" also seems to accept the `start_up_on_off` value: 
![ZHA screen in Home Assistant showing the different switches of the board with different power-on states](https://github.com/user-attachments/assets/0ecc88d5-684d-4304-aa36-bae8d6613c88)
![THA screen of Home Assistant showing the 0x4003 property of the 0x0006 cluster endpoint 2](https://github.com/user-attachments/assets/8a7dabd8-a7a8-4eea-a10b-07f6b23e1ef0)
Unfortunately, I cannot test whether this actually works right now, because the plug is neatly hidden behind a cabinet. It does look like the values are mapped and accepted by the device, though.

As for the `child_lock`, my power strip doesn't accept setting them at all, and only returns `None`. To be honest, that sounds about right, though. The `child_lock` property seems to be displayed for each switch, but if it was actually supported/worked, I'd expect there to be one `child_lock` setting for the main power button, and the main button as a separate cluster. Having different child lock settings for each socket doesn't make too much sense

hangy committed:
Add quirk for TZ3000

Adds support for UseeLink 302EU-Zigbee, based on https://github.com/zigpy/zha-device-handlers/issues/1632#issuecomment-1627072807

Co-authored-by: ingoratsdorf <ingoratsdorf@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
